demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,874 @@
+        {
+            LogError( ( "Failed to subscribe to fleet provisioning topic: %.*s.",
+                        FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                        FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool unsubscribeFromRegisterThingResponseTopics( void )
+{
+    bool status;
+
+    status = UnsubscribeFromTopic( FP_CBOR_REGISTER_ACCEPTED_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                                   FP_CBOR_REGISTER_ACCEPTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+    if( status == false )
+    {
+        LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                    FP_CBOR_REGISTER_ACCEPTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                    FP_CBOR_REGISTER_ACCEPTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+    }
+
+    if( status == true )
+    {
+        status = UnsubscribeFromTopic( FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                                       FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+        if( status == false )
+        {
+            LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                        FP_CBOR_REGISTER_REJECTED_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                        FP_CBOR_REGISTER_REJECTED_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool saveCertificate( const char * pCertificate,
+                             size_t certificateLength )
+{
+    FILE * file;
+    bool status = false;
+
+    file = fopen( PROVISIONING_CERT_PATH, "wb" );
+
+    if( file == NULL )
+    {
+        LogError( ( "Failed opening PROVISIONING_CERT_PATH for writing." ) );
+    }
+    else
+    {
+        size_t written;
+        written = fwrite( ( const void * ) pCertificate, 1U, certificateLength, file );
+        fclose( file );
+
+        if( written != certificateLength )
+        {
+            LogError( ( "Failed writing certificate to file." ) );
+        }
+        else
+        {
+            status = true;
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+/* This example uses a single application task, which shows that how to use
+ * the Fleet Provisioning library to generate and validate AWS IoT Fleet
+ * Provisioning MQTT topics, and use the coreMQTT library to communicate with
+ * the AWS IoT Fleet Provisioning APIs. */
+int main( int argc,
+          char ** argv )
+{
+    bool status = false;
+    int exitStatus = EXIT_FAILURE;
+    char serial[ 10 ];
+    size_t serialLength = 10;
+    /* Buffer for holding the CSR. */
+    char csr[ NETWORK_BUFFER_SIZE ] = { 0 };
```

Comment:
Why is the CSR buffer size as huge as the the buffer size of the MQTT context's network buffer size? The CSR is one part of the entire network buffer size required for MQTT communication...